crates/forge/bin/cmd/test/mod.rs

```diff
@@ -358,11 +369,54 @@ impl TestArgs {
             "decode-internal",
             self.decode_internal.as_ref().and_then(|v| v.as_ref()),
         )?;
+        maybe_override_mt("flamegraph", self.flamegraph.as_ref())?;
 
         let libraries = runner.libraries.clone();
-        let outcome = self.run_tests(runner, config, verbosity, &filter, &output).await?;
+        let mut outcome = self.run_tests(runner, config, verbosity, &filter, &output).await?;
+
+        if should_flamegraph {
+            let test_result = outcome
+                .results
+                .iter_mut()
+                .find(|(_, r)| !r.test_results.is_empty())
+                .map(|(_, r)| (r.test_results.values_mut().next().unwrap()))
+                .unwrap();
+
+            let arena = test_result
+                .traces
+                .iter_mut()
+                .find_map(
+                    |(kind, arena)| {
+                        if *kind == TraceKind::Execution {
+                            Some(arena)
+                        } else {
+                            None
+                        }
+                    },
+                )
+                .unwrap();
+
+            let decoder = outcome.last_run_decoder.as_ref().unwrap();
+
+            decode_trace_arena(arena, decoder).await?;
+            let mut fst = folded_stack_trace::build(arena);
+            fst.reverse();
+
+            let file_name = "flamegraph.svg";
+            let writer = std::fs::File::create(file_name).unwrap();
+
+            let mut options = inferno::flamegraph::Options::default();
+            options.flame_chart = true;
+
+            inferno::flamegraph::from_lines(&mut options, fst.iter().map(|s| s.as_str()), writer)
```

Comment:
Instead of writing to file, reading it and writing again, you can use `Vec<u8>` or any other in-memory type which implements `Write`